source/core_pkcs11.c

```diff
@@ -319,7 +319,7 @@ CK_RV xFindObjectWithLabelAndClass( CK_SESSION_HANDLE xSession,
     {
         xTemplate[ 0 ].type = CKA_LABEL;
         xTemplate[ 0 ].pValue = ( CK_VOID_PTR ) pcLabelName;
-        xTemplate[ 0 ].ulValueLen = strnlen( pcLabelName, pkcs11configMAX_LABEL_LENGTH );
+        xTemplate[ 0 ].ulValueLen = strlen( pcLabelName );
```

Comment:
this is better with the length, why are we removing it?

Reply:
strnlen is not in C90, but is a POSIX function

Reply:
At least 3 vendors have a stdlib that does not implement it

Reply:
I did not realize this, when I initially made this change.

Reply:
strlen is not MISRA compliant ...

Reply:
I think the best solution is to change the API to pass the label length in to the library.